source-hive/src/main/java/org/apache/kylin/source/hive/MRHiveDictUtil.java

```diff
@@ -114,33 +122,39 @@ public static String generateDropTableStatement(String tableName) {
     }
 
     public static String generateDistinctValueTableStatement(IJoinedFlatTableDesc flatDesc) {
-        StringBuilder ddl = new StringBuilder();
-        String table = flatDesc.getTableName()
-                + flatDesc.getSegment().getConfig().getMrHiveDistinctValueTableSuffix();
+        KylinConfig config = flatDesc.getSegment().getConfig();
+        String table = config.getMrHiveDistinctValueTableSuffix();
```

Comment:
`table` should be `flatDesc.getTableName() + config().getMrHiveDistinctValueTableSuffix()`